spinalcordtoolbox/scripts/sct_download_data.py

```diff
@@ -76,7 +76,9 @@
     },
     "binaries_linux": {
         "mirrors": [
-            "https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries/releases/download/r20220516-2/spinalcordtoolbox-binaries_linux.tar.gz",
+            #  "https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries/releases/download/r20220516-2/spinalcordtoolbox-binaries_linux.tar.gz",
+            #  only for testing purposes
+            "https://syncandshare.lrz.de/dl/fi9eCRkTSJBH55a6qmpXc2k7/spinalcordtoolbox-binaries_linux.tar.gz",
```

Comment:
When I run the following, as per https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3865/files#r955854012:

```
sct_image -i sct_roi1.nii.gz sct_roi2.nii.gz -o stitched.nii.gz -stitch
```

I get the following error:


```
RuntimeError: /home/joshua/repos/spinalcordtoolbox/bin/stitching: error while loading shared libraries: libboost_program_options.so.1.71.0: cannot open shared object file: No such file or directory
```

I believe this means that the `stitching` binary was built using [dynamic linking, as opposed to static linking](https://pediaa.com/what-is-the-difference-between-static-and-dynamic-linking/). (See also: [this SO answer](https://stackoverflow.com/a/725550).)

I'm a little shaky when it comes to C++, but I'll keep digging into this, especially as part of https://github.com/jqmcginnis/build_biomedia_stitching/pull/1.

Reply:
I think I have a working statically-linked `stitching` binary: https://github.com/jqmcginnis/build_biomedia_stitching/pull/1#discussion_r956287929

So, I've uploaded this binary to https://github.com/jqmcginnis/build_biomedia_stitching/releases/tag/temp-release, and updated the link to test it in this PR. :slightly_smiling_face:

Reply:
Aha, now we get the following error (note the different between `1.71.0` and `1.80.0`):

```
 RuntimeError: /__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching: error while loading shared libraries: libboost_program_options.so.1.80.0: cannot open shared object file: No such file or directory
```

It looks like I had fooled myself into thinking that I had done static linking, when in fact I had done dynamic linking. I'll try again. :)

Reply:
Ahh, TIL: You can use `ldd` to check this:

```
joshua@tadpole:~/Desktop$ ldd stitching 
	linux-vdso.so.1 (0x00007ffffebf5000)
	libboost_program_options.so.1.80.0 => /home/joshua/Desktop/build_biomedia_stitching/3rdParty/boost_1_80_0/stage/lib/libboost_program_options.so.1.80.0 (0x00007f3734991000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f37348aa000)
	libboost_filesystem.so.1.80.0 => /home/joshua/Desktop/build_biomedia_stitching/3rdParty/boost_1_80_0/stage/lib/libboost_filesystem.so.1.80.0 (0x00007f3735940000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f3734600000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f373488a000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f3734200000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f373597a000)
```

Reply:
I think I've found a solution that removed the dynamic linking to the Boost libraries. 

If you set `set(Boost_USE_STATIC_LIBS ON)` inside of the following files:

* `itkio/cmake/FindConfig.cmake`
* `stitching/cmake/FindConfig.cmake`

The resulting built binary will have the following `ldd` output:

```console
joshua@tadpole:~/Desktop$ ldd stitching 
        linux-vdso.so.1 (0x00007fffeddd6000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f0fd5a0e000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f0fd4600000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f0fd59ee000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f0fd4200000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f0fd5b08000)
```

Reply:
Despite eliminating the dynamic linking to Boost, the CI is now reporting these new errors:

```
 RuntimeError: /__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching: /lib64/libm.so.6: version `GLIBC_2.29' not found (required by /__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching)
/__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching)
/__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by /__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching)
/__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching)
/__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching)
/__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /__w/spinalcordtoolbox/spinalcordtoolbox/bin/stitching)
```

I think that these are due to the fact that I've compiled `stitching` on my local Ubuntu 22.04 machine, which dynamically links to a version of `glibc` and `glibcxx` too new for the GitHub Actions runners.

(In this case, [dynamic linking for glibc is OK](https://stackoverflow.com/questions/57476533/why-is-statically-linking-glibc-discouraged), I think. Our ANTs binaries are dynamically linked to the exact same `.so` files. It's just my environment is too new.)

I'm... somewhat confident that building for, say, `ubuntu-18.04` will help ensure compatibility, since `glibc` updates [tend to be backwards compatible](https://superuser.com/a/1450958)? But, I'm also still shaky with this subject, so I'm sort of learning as I go here.

Are due to the fact that

Reply:
Getting closer! I finally got the `ubuntu-20.04` automated build of `stitching` [working](https://github.com/jqmcginnis/build_biomedia_stitching/actions/runs/2952262425). Using that `stitching` binary, (as expected) the Ubuntu 20.04 build passes. But, I'm still getting some `glibc` errors from the CentOS Stream 8 build. So, I'm going to try downgrading the runner as far as I can. :)

Reply:
There we are! Building with [`ubuntu-18.04`](https://github.com/jqmcginnis/build_biomedia_stitching/actions/runs/2952464939) caused the [tests to pass](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/8082169792).